src/manifolds/PositiveNumbers.jl

```diff
@@ -0,0 +1,224 @@
+
+Check whether `X` is a tangent vector in the tangent space of `p` on the
+[`PositiveNumbers`](@ref) `M`.
+For the real-valued case represented by positive numbers, all `X` are valid, since the tangent space is the whole real line.
+For the complex-valued case `X` [...]
+
+"""
+function check_tangent_vector(M::PositiveNumbers, p, X; check_base_point = true, kwargs...)
+    if check_base_point
+        perr = check_manifold_point(M, p; kwargs...)
+        return perr # if x is valid all v that are real numbers are valid
+    end
+    return nothing
+end
+
+@doc raw"""
+    distance(M::PositiveNumbers, p, q)
+
+Compute the distance on the [`PositiveNumbers`](@ref) `M`, which is
+
+````math
+d(p,q) = \Bigl\lvert \log \frac{p}{q} \Bigr\rvert = \lvert \log p - \log q\rvert.
+````
+"""
+distance(::PositiveNumbers, p::Real, q::Real) = abs(log(p / q))
+
+@doc raw"""
+    exp(M::PositiveNumbers, p, X)
+
+Compute the exponential map on the [`PositiveNumbers`](@ref) `M`.
+```math
+\exp_p X = p\operatorname{exp}(X/p).
+```
+"""
+Base.exp(::PositiveNumbers, p::Real, X::Real) = p * exp(X / p)
+
+flat(::PositiveNumbers, ::Number, X::TFVector) = FVector(CotangentSpace, X.data)
+
+@doc raw"""
+    injectivity_radius(M::PositiveNumbers[, p])
+
+Return the injectivity radius on the [`PositiveNumbers`](@ref) `M`, i.e. $\infty$.
+"""
+injectivity_radius(::PositiveNumbers) = Inf
+injectivity_radius(::PositiveNumbers, ::ExponentialRetraction) = Inf
+injectivity_radius(::PositiveNumbers, ::Any) = Inf
+injectivity_radius(::PositiveNumbers, ::Any, ::ExponentialRetraction) = Inf
+eval(
+    quote
+        @invoke_maker 1 Manifold injectivity_radius(
+            M::PositiveNumbers,
+            rm::AbstractRetractionMethod,
+        )
+    end,
+)
+
+@doc raw"""
+    inner(M::PositiveNumbers, p, X, Y)
+
+Compute the inner product of the two tangent vectors `X,Y` from the tangent plane at `p` on
+the [`PositiveNumbers`](@ref) `M`, i.e.
+
+````math
+g_p(X,Y) = \frac{XY}{p^2}.
+````
+"""
+inner(::PositiveNumbers, ::Any...)
+@inline inner(::PositiveNumbers, p::Real, X::Real, Y::Real) = X * Y / p^2
+
+function inverse_retract(M::PositiveNumbers, x::Number, y::Number)
+    return inverse_retract(M, x, y, LogarithmicInverseRetraction())
+end
+function inverse_retract(
+    M::PositiveNumbers,
+    x::Number,
+    y::Number,
+    ::LogarithmicInverseRetraction,
+)
+    return log(M, x, y)
+end
+
+@doc raw"""
+    log(M::PositiveNumbers, p, q)
+
+Compute the logarithmic map on the [`PositiveNumbers`](@ref) `M`.
+````math
+\log_p q = p\log\frac{q}{p}.
+````
+"""
+Base.log(::PositiveNumbers, p::Real, q::Real) = p * log(q / p)
+
+@doc raw"""
+    manifold_dimension(M::PositiveNumbers)
+
+Return the dimension of the [`PositiveNumbers`](@ref) `M`,
+i.e. of the 1-dimensional hyperbolic space,
+
+````math
+\dim(H^1) = 1
+````
+"""
+manifold_dimension(::PositiveNumbers) = 1
+
+mid_point(M::PositiveNumbers, p1, p2) = exp(M, p1, 0.5 * log(M, p1, p2))
```

Comment:
```suggestion
mid_point(M::PositiveNumbers, p1, p2) = exp(M, p1, log(M, p1, p2) / 2)
```
This won't force type promotion to at least `Float64`.